src/interactivity/win32/Clipboard.cpp

```diff
@@ -52,35 +52,35 @@ contents and writing them to the console's input buffer
 --*/
 void Clipboard::Paste()
 {
-    HANDLE ClipboardDataHandle;
-
-    // Clear any selection or scrolling that may be active.
-    Selection::Instance().ClearSelection();
-    Scrolling::s_ClearScroll();
+    // WIP auditing if user is enrolled
+    static auto DestinationName = _LoadString(ID_CONSOLE_WIP_DESTINATIONNAME);
+    Microsoft::Console::Internal::EdpPolicy::AuditClipboard(DestinationName);
 
-    // Get paste data from clipboard
-    if (!OpenClipboard(ServiceLocator::LocateConsoleWindow()->GetWindowHandle()))
+    const auto clipboard = _openClipboard(ServiceLocator::LocateConsoleWindow()->GetWindowHandle());
+    if (!clipboard)
     {
+        LOG_LAST_ERROR();
         return;
     }
 
-    ClipboardDataHandle = GetClipboardData(CF_UNICODETEXT);
-    if (ClipboardDataHandle == nullptr)
+    const auto handle = GetClipboardData(CF_UNICODETEXT);
+    if (!handle)
     {
-        CloseClipboard();
         return;
     }
 
-    auto pwstr = (PWCHAR)GlobalLock(ClipboardDataHandle);
-    StringPaste(pwstr, (ULONG)GlobalSize(ClipboardDataHandle) / sizeof(WCHAR));
-
-    // WIP auditing if user is enrolled
-    static auto DestinationName = _LoadString(ID_CONSOLE_WIP_DESTINATIONNAME);
-    Microsoft::Console::Internal::EdpPolicy::AuditClipboard(DestinationName);
-
-    GlobalUnlock(ClipboardDataHandle);
+    // Clear any selection or scrolling that may be active.
+    Selection::Instance().ClearSelection();
+    Scrolling::s_ClearScroll();
 
-    CloseClipboard();
+    const auto str = static_cast<const wchar_t*>(GlobalLock(handle));

```

Comment:
wil may have a HGLOBAL helper for this

Reply:
or maybe you already knew that b/c i see you used one later

Reply:
The one in wil is `void*` which wasn't that nice to use. In this case, the `StringPaste` call won't throw so the code is fine IMO. But looking at it again, I would indeed feel more comfortable with a `scope_exit` at least...